ambry-router/src/test/java/com.github.ambry.router/PutOperationTest.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.commons.ByteBufferReadableStreamChannel;
+import com.github.ambry.commons.ResponseHandler;
+import com.github.ambry.config.RouterConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.messageformat.BlobProperties;
+import com.github.ambry.network.NetworkReceive;
+import com.github.ambry.network.RequestInfo;
+import com.github.ambry.network.ResponseInfo;
+import com.github.ambry.protocol.PutRequest;
+import com.github.ambry.protocol.RequestOrResponse;
+import com.github.ambry.utils.ByteBufferChannel;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import java.util.TreeMap;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class PutOperationTest {
+  private final RouterConfig routerConfig;
+  private final MockClusterMap mockClusterMap;
+  private final ResponseHandler responseHandler;
+  private final Time time;
+  private final Map<Integer, PutOperation> correlationIdToPutOperation = new TreeMap<>();
+  private final MockServer mockServer = new MockServer();
+
+  private class PutTestRequestRegistrationCallbackImpl implements PutRequestRegistrationCallback {
```

Comment:
You will have some fixing to do depending on the order your PRs go in because you have changed this. Just a heads up because I doubt git will treat this as a conflict. The build will fail *after* the merge to upstream if we don't fix it.